packages/lumx-react/src/components/tabs/TabList.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { Children, ReactElement, ReactNode, cloneElement } from 'react';
+
+import classNames from 'classnames';
+
+import { Theme } from '@lumx/react';
+import { COMPONENT_PREFIX, CSS_PREFIX } from '@lumx/react/constants';
+import { GenericProps, handleBasicClasses } from '@lumx/react/utils';
+import { useRovingTabIndex } from '../../hooks/useRovingTabIndex';
+
+enum TabListLayout {
+    clustered = 'clustered',
+    fixed = 'fixed',
+}
+
+enum TabListPosition {
+    center = 'center',
+    left = 'left',
+    right = 'right',
+}
+
+/**
+ * Defines the props of the component.
+ */
+interface TabListProps extends GenericProps {
+    /** The label that describes the purpose of the set of tabs. */
+    ['aria-label']: string;
+    /** The active tab. */
+    activeTabId?: string;
+    /** The children elements. */
+    children: ReactNode;
+    /** The layout of the tabs. */
+    layout?: TabListLayout;
+    /** The position of the tabs. */
+    position?: TabListPosition;
+    /** The theme to apply to the component. Can be either 'light' or 'dark'. */
+    theme?: Theme;
+    /** Whether custom colors are applied to this component or not. */
+    useCustomColors?: boolean;
+    /** The function called on tab change. */
+    onChange(id: string): void;
+    /** The function called on tab focus. */
+    onFocus?(id: string): void;
+}
+
+/**
+ * The display name of the component.
+ */
+const COMPONENT_NAME = `${COMPONENT_PREFIX}TabList`;
+
+/**
+ * The default class name and classes prefix for this component.
+ */
+const CLASSNAME = `${CSS_PREFIX}-tabs`;
+
+/**
+ * The default value of props.
+ */
+const DEFAULT_PROPS: Partial<TabListProps> = {
+    layout: TabListLayout.fixed,
+    position: TabListPosition.left,
+    theme: Theme.light,
+};
+
+const TabList: React.FC<TabListProps> = ({
+    activeTabId,
+    ['aria-label']: ariaLabel,
+    children,
+    className,
+    layout,
+    onChange,
+    onFocus,
+    position,
+    theme,
+    useCustomColors,
+    ...forwardedProps
+}) => {
+    const tabListRef = React.useRef(null);
+    useRovingTabIndex({ parentRef: tabListRef, elementSelector: '[role="tab"]', keepTabIndex: false });
+
+    const tabs: ReactElement[] = Children.map(children as ReactElement[], (tab: ReactElement, index: number) => {
+        /** If no activeTabId, the first tab is considered active. */
+        const isActive = activeTabId === tab.props.id || (!activeTabId && index === 0);
+
+        return cloneElement(tab, {
+            id: tab.props.id as string,
+            isActive,
+            /** Tabs should only have one focusable tab at a time. It should be the active tab. */
+            tabIndex: isActive ? 0 : -1,
+            onChange,
+            onFocus,
+        });
+    });
```

Comment:
I realised we forgot about blocking selection when a tab is disabled :grimacing: 

Could we maybe do something like this ?
```suggestion
    const tabs: ReactElement[] = Children.map(children as ReactElement[], (tab: ReactElement, index: number) => {
        const { id: tabId, isDisabled } = tab.props;
        /** If no activeTabId, the first tab is considered active. */
        const isActive = activeTabId === tabId || (!activeTabId && index === 0);

        return cloneElement(tab, {
            id: tabId as string,
            isActive,
            /** Tabs should only have one focusable tab at a time. It should be the active tab. */
            tabIndex: isActive ? 0 : -1,
            onChange: !isDisabled && onChange,
            onFocus: !isDisabled && onFocus,
        });
    });
```